diff --git a/dlls/sxs/sxs.c b/dlls/sxs/sxs.c
index 150a49a322f..a7aa4f630a7 100644
--- a/dlls/sxs/sxs.c
+++ b/dlls/sxs/sxs.c
@@ -24,6 +24,8 @@
 #include "windef.h"
 #include "winbase.h"
 
+#include "wine/exception.h"
+
 #include "wine/debug.h"
 
 WINE_DEFAULT_DEBUG_CHANNEL(sxs);
@@ -199,21 +201,42 @@ BOOL WINAPI SxsLookupClrGuid(DWORD flags, GUID *clsid, HANDLE actctx, void *buff
     ret->pcwszAssemblyIdentity = (WCHAR *)ret_strings;
     ret_strings += len_identity;
 
+    /* FIXME the offsets like in ptr_name = (char *)surrogate + surrogate->name_offset; (line 175) seem to contain bogus values ???? */
+    __TRY 
+    {
     memcpy(ret_strings, ptr_name, len_name);
     ret->pcwszTypeName = (WCHAR *)ret_strings;
     ret_strings += len_name;
+    }
+    __EXCEPT_PAGE_FAULT
+    {
+	ret->pcwszTypeName=0;	
+	}
+    __ENDTRY
+
+
+
 
     if (len_version > 0)
-    {
+    {   __TRY 
+		{
         memcpy(ret_strings, ptr_version, len_version);
         ret->pcwszRuntimeVersion = (WCHAR *)ret_strings;
+	    }
+	    __EXCEPT_PAGE_FAULT
+	    {
+	    ret->pcwszRuntimeVersion = NULL;
+	    }
+	    __ENDTRY
     }
     else
         ret->pcwszRuntimeVersion = NULL;
 
+
+
     SetLastError(0);
 
-out:
+out: if(retval) FIXME("version /identity /name = %s, %s, %s\n", wine_dbgstr_w(ret->pcwszRuntimeVersion), wine_dbgstr_w(ret->pcwszAssemblyIdentity), wine_dbgstr_w(ret->pcwszTypeName));
     ReleaseActCtx(guid_info.hActCtx);
 
     if (flags & SXS_LOOKUP_CLR_GUID_USE_ACTCTX)
