diff --git a/dlls/cfgmgr32/main.c b/dlls/cfgmgr32/main.c
index b545d761f68..fdc3e0d98c3 100644
--- a/dlls/cfgmgr32/main.c
+++ b/dlls/cfgmgr32/main.c
@@ -66,7 +66,9 @@ CONFIGRET WINAPI CM_Register_Notification( CM_NOTIFY_FILTER *filter, void *conte
 {
     FIXME("%p %p %p %p stub!\n", filter, context, callback, notify_context);
 
-    return CR_CALL_NOT_IMPLEMENTED;
+    *notify_context = CreateEventW(NULL, TRUE, FALSE, NULL);
+ 
+    return 0;
 }
 
 /***********************************************************************
