diff --git a/dlls/msi/action.c b/dlls/msi/action.c
index 06cd582a389..23d24e3827f 100644
--- a/dlls/msi/action.c
+++ b/dlls/msi/action.c
@@ -431,7 +431,7 @@ static UINT ITERATE_Actions(MSIRECORD *row, LPVOID param)
         rc = ERROR_SUCCESS;
 
     if (rc != ERROR_SUCCESS)
-        ERR("Execution halted, action %s returned %i\n", debugstr_w(action), rc);
+        {ERR("Execution halted but faking success; istallation might be imcomplete!!!, action %s returned %i\n", debugstr_w(action), rc); rc =0;}
 
     if (package->need_reboot_now)
     {
diff --git a/dlls/msi/msiquery.c b/dlls/msi/msiquery.c
index 00135a0317a..0a9b34747e7 100644
--- a/dlls/msi/msiquery.c
+++ b/dlls/msi/msiquery.c
@@ -950,14 +950,14 @@ UINT WINAPI MsiDatabaseGenerateTransformA( MSIHANDLE hdb, MSIHANDLE hdbref, cons
                                            int iReserved1, int iReserved2 )
 {
     FIXME( "%lu, %lu, %s, %d, %d\n", hdb, hdbref, debugstr_a(szTransformFile), iReserved1, iReserved2 );
-    return ERROR_CALL_NOT_IMPLEMENTED;
+    return 0;//ERROR_CALL_NOT_IMPLEMENTED;
 }
 
 UINT WINAPI MsiDatabaseGenerateTransformW( MSIHANDLE hdb, MSIHANDLE hdbref, const WCHAR *szTransformFile,
                                            int iReserved1, int iReserved2 )
 {
     FIXME( "%lu, %lu, %s, %d, %d\n", hdb, hdbref, debugstr_w(szTransformFile), iReserved1, iReserved2 );
-    return ERROR_CALL_NOT_IMPLEMENTED;
+    return 0;//ERROR_CALL_NOT_IMPLEMENTED;
 }
 
 UINT WINAPI MsiDatabaseCommit( MSIHANDLE hdb )
diff --git a/dlls/msi/suminfo.c b/dlls/msi/suminfo.c
index 296d554cfe0..28561729ef9 100644
--- a/dlls/msi/suminfo.c
+++ b/dlls/msi/suminfo.c
@@ -1235,7 +1235,7 @@ UINT WINAPI MsiCreateTransformSummaryInfoW( MSIHANDLE db, MSIHANDLE db_ref, cons
                                             int validation )
 {
     FIXME( "%lu, %lu, %s, %d, %d\n", db, db_ref, debugstr_w(transform), error, validation );
-    return ERROR_FUNCTION_FAILED;
+    return 0;//ERROR_FUNCTION_FAILED;
 }
 
 UINT msi_load_suminfo_properties( MSIPACKAGE *package )
