diff --git a/dlls/rpcrt4/cpsf.c b/dlls/rpcrt4/cpsf.c
index 006e6bce62f..e34969ef898 100644
--- a/dlls/rpcrt4/cpsf.c
+++ b/dlls/rpcrt4/cpsf.c
@@ -185,6 +185,10 @@ HRESULT WINAPI NdrDllGetClassObject(REFCLSID rclsid, REFIID iid, LPVOID *ppv,
     debugstr_guid(iid), ppv, pProxyFileList, debugstr_guid(pclsid),
     pPSFactoryBuffer);
 
+  char *clsid = strdup(debugstr_guid(rclsid));
+
+  if(!strncmp(clsid,"{88d96a",7) && !strncmp(clsid+9,"-f192-11d4-a65f-0040963251e5}",29)) return E_NOTIMPL;
+
   *ppv = NULL;
   if (!pPSFactoryBuffer->lpVtbl) init_psfactory( pPSFactoryBuffer, pProxyFileList );
 
