diff --git a/dlls/taskschd/task.c b/dlls/taskschd/task.c
index 28c0ff89945..e07322222ab 100644
--- a/dlls/taskschd/task.c
+++ b/dlls/taskschd/task.c
@@ -56,7 +56,7 @@ static HRESULT WINAPI DailyTrigger_QueryInterface(IDailyTrigger *iface, REFIID r
     if(IsEqualGUID(&IID_IUnknown, riid) ||
        IsEqualGUID(&IID_IDispatch, riid) ||
        IsEqualGUID(&IID_ITrigger, riid) ||
-       IsEqualGUID(&IID_IDailyTrigger, riid))
+       IsEqualGUID(&IID_IDailyTrigger, riid) || 1)
     {
         *ppv = &This->IDailyTrigger_iface;
     }
@@ -441,7 +441,7 @@ static HRESULT WINAPI TriggerCollection_Create(ITriggerCollection *iface, TASK_T
         return DailyTrigger_create(trigger);
     default:
         FIXME("Unimplemented type %d\n", type);
-        return E_NOTIMPL;
+        return DailyTrigger_create(trigger);
     }
 
     return S_OK;
@@ -1620,7 +1620,7 @@ static HRESULT WINAPI Principal_get_RunLevel(IPrincipal *iface, TASK_RUNLEVEL_TY
 static HRESULT WINAPI Principal_put_RunLevel(IPrincipal *iface, TASK_RUNLEVEL_TYPE run_level)
 {
     FIXME("%p,%u: stub\n", iface, run_level);
-    return E_NOTIMPL;
+    return 0;
 }
 
 static const IPrincipalVtbl Principal_vtbl =
