diff --git a/dlls/sxs/sxs.c b/dlls/sxs/sxs.c
index 150a49a322f..37c18ab08ef 100644
--- a/dlls/sxs/sxs.c
+++ b/dlls/sxs/sxs.c
@@ -104,9 +104,19 @@ BOOL WINAPI SxsLookupClrGuid(DWORD flags, GUID *clsid, HANDLE actctx, void *buff
     char *ret_strings;
     ULONG_PTR cookie;
 
-    TRACE("%#lx, %s, %p, %p, %Ix, %p.\n", flags, wine_dbgstr_guid(clsid), actctx,
+    FIXME("%#lx, %s, %p, %p, %Ix, %p.\n", flags, wine_dbgstr_guid(clsid), actctx,
           buffer, buffer_len, buffer_len_required);
 
+    /*** FIXME Stupid hacks below for Visual Studio 2022 ***/
+
+    if(!strcmp(wine_dbgstr_guid(clsid), "{1ded92b5-9a46-4b29-93ef-b5e07016659e}")) {return 0;}
+    if(!strcmp(wine_dbgstr_guid(clsid), "{88ac98c7-b38c-404b-bd86-d2a4f2e89dca}")) {return 0;}
+    if(!strcmp(wine_dbgstr_guid(clsid), "{d83d60e3-229f-4660-8dd0-28b629eedcda}")) {return 0;}
+    if(!strcmp(wine_dbgstr_guid(clsid), "{45fb4600-e6e8-4928-b25e-50476ff79425}")) {return 0;}
+    if(!strcmp(wine_dbgstr_guid(clsid), "{177f0c4a-1cd3-4de7-a32c-71dbbb9fa36d}")) {return 0;}
+    if(!strcmp(wine_dbgstr_guid(clsid), "{f5e7e71e-1401-11d1-883b-0000f87579d2}")) {return 0;}
+    
+    
     if (flags & SXS_LOOKUP_CLR_GUID_USE_ACTCTX)
     {
         if (!ActivateActCtx(actctx, &cookie))
